vowpalwabbit/options.h

```diff
@@ -193,6 +183,122 @@ struct options_i
   virtual ~options_i() = default;
 };
 
+struct option_group_definition
+{
+  // add second parameter for const string short name
+  option_group_definition(const std::string& name) : m_name(name) {}
+
+  template <typename T>
+  option_group_definition& add(T&& op)
+  {
+    m_options.push_back(std::make_shared<typename std::decay<T>::type>(op));
+    return *this;
+  }
+
+  template <typename T>
+  option_group_definition& add(typed_option<T>& op)
+  {
+    m_options.push_back(std::make_shared<typename std::decay<typed_option<T>>::type>(op));
+    if (op.m_necessary) { m_necessary_flags.insert(op.m_name); }
+
+    return *this;
+  }
+
+  // will check if ALL of 'necessary' options were suplied
+  bool check_necessary_enabled(const options_i& options) const

```

Comment:
nit: should this be named `check_necessary_supplied`?

Reply:
I see both equally the same, what's making you opt for supplied? the comment above or you think it's more ad-hoc for what's going on?

Reply:
The comment above got me thinking. This is a super super nit but semantically we could say that the user supplied an option and we enable the reduction. I am happy with either naming